### PR TITLE
CPLAT-8261: Post-Dart2 cleanup

### PR DESCRIPTION
**Pull request by olesiathoms-wk**
### Description
This PR performs dart1 removal.

### Changes
- `dartfmt` fixes
- `dart2_constant` was removed
- updated dependencies
- remove deprecated lint rules
- remove unused files
- run `.travis.yaml` on dart `2.4.0`, `stable` and `dev`
- cleanup `.gitignore`
- use `isA` instead of `isInstanceOf`
- remove `typed()` from `mockito`

### Testing instructions
- [ ] CI should be sufficient

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
-  - stable
-
-script:
-  - tool/format.sh
-  - dartanalyzer lib bin test/unit
-  - pub publish --dry-run
-  - xvfb-run --server-args="-screen 0 1024x768x24" tool/test.sh
+jobs:
+  include:
+    - dart: 2.4.0
```
Comment (smaifullerton-wk) [inline .travis.yml]: Same thing - wanna use dart 2.6.1 here?

Comment (olesiathoms-wk) [reply .travis.yml]: we already have `stable` that will run 2.6.1 for us
https://travis-ci.org/Workiva/w_common/jobs/611587650#L189

Comment (smaifullerton-wk) [reply .travis.yml]: ohhh

**Comment by smaifullerton-wk** [comment]
+10 CI passes

**Comment by smaifullerton-wk** [comment]
@Workiva/release-management-p

**Pull request merged**